### PR TITLE
[struct_json] is compatible with msvc versions

### PR DESCRIPTION
**Pull request by RAVER0306**
The purpose of this change is to ensure compatibility with different versions of the Microsoft Visual Studio (MSVC) compiler. The `_MSC_VER` macro is used to determine the compiler version, allowing the code to handle differences between Visual Studio 2019 (MSVC 14.2x) and Visual Studio 2022 (MSVC 14.30+). This is particularly important because VS2022 introduces full support for C++20 features, while earlier versions require a fallback implementation.

## What is changing

The code now includes conditional checks for `_MSC_VER`:
- When `_MSC_VER >= 1930` (Visual Studio 2022 or newer), the code leverages C++20 features such as generic lambdas and fold expressions.
- For older versions like VS2019 (`_MSC_VER >= 1920 && _MSC_VER < 1930`), the code falls back to an alternative implementation to maintain compatibility.

This ensures that the code can be compiled with both VS2019 and VS2022 while taking full advantage of C++20 features where available.

## Example

For example, the following macro-based logic ensures that the code behaves correctly in both compilers:

```cpp
#if __cplusplus >= 202002L
  #if _MSC_VER >= 1930
    // Code that uses C++20 features, supported in VS2022
  #else
    // Fallback code for VS2019, which lacks full C++20 support
  #endif
#else
  // Code for non-C++20 environments
#endif

### CODE REVIEW

File: include/ylt/reflection/member_names.hpp [hunk 1]
```diff
@@ -127,13 +127,25 @@ get_member_names() {
...
+    init_arr_with_tuple<T>(arr, std::make_index_sequence<Count>{});
+#endif
+#else
+    constexpr auto tp = struct_to_tuple<T>();
+    [&]<size_t... Is>(std::index_sequence<Is...>) mutable {
+      ((arr[Is] =
+            internal::get_member_name<internal::wrap(std::get<Is>(tp))>()),
+       ...);
+    }(std::make_index_sequence<Count>{});
+#endif
```
Comment (qicosmos) [inline include/ylt/reflection/member_names.hpp:148]: 这样改有点啰嗦，看能不能这样改：
```cpp
#if (defined(_MSC_VER) &&  _MSC_VER < 1930) || __cplusplus < 202002L
  init_arr_with_tuple<T>(arr, std::make_index_sequence<Count>{});
#else
    constexpr auto tp = struct_to_tuple<T>();
    [&]<size_t... Is>(std::index_sequence<Is...>) mutable {
      ((arr[Is] =
            internal::get_member_name<internal::wrap(std::get<Is>(tp))>()),
       ...);
    }
    (std::make_index_sequence<Count>{});
#endif
```

其它的地方也是类似的改法。

Comment (RAVER0306) [reply include/ylt/reflection/member_names.hpp]: 已简化

**Comment by qicosmos** [comment]
image
按照+号的缩进，手动编辑一下吧

**Comment by RAVER0306** [comment]
> image 按照+号的缩进，手动编辑一下吧

**Pull request merged**